src/app/Controllers/BaseController.cs

```diff
@@ -84,5 +98,29 @@ public static JsonResult CreateResult(string message, HttpStatusCode statusCode)
                 StatusCode = (int)statusCode
             };
         }
+
+        /// <summary>
+        /// This method creates a Polly retry policy when there is cosmos exception with code Unauthorized.
+        /// </summary>
+        /// <returns></returns>
+        private AsyncRetryPolicy GetCosmosRetryPolicy()
+        {
+            return Policy.Handle<CosmosException>(e => e.StatusCode == HttpStatusCode.Unauthorized)
+                .WaitAndRetryAsync(1, retryAttempt =>
+                 {
+                     var timeToWait = TimeSpan.FromSeconds(Math.Pow(2, retryAttempt));
+                     logger.LogInformation("Reloading configurations to check if Key was rotated.");
+                     
+                     //Reload the configurations.
+                     (configuration as IConfigurationRoot).Reload();
```

Comment:
This is an expensive operation and needs to be encapsulated so every request is not running a reload (which is n+1 KeyVault requests with n being the number of secrets in KV). 

At scale, the key vault requests will get throttled which will result in errors.

Reply:
@bartr,  We can refresh the connection without reloading the configurations using IKeyVaultConnection but in that case our configuration store will still be having the old key while Cosmos dal connection will be using the new Key.
So it is just the keys will be out of sync until the app restarts.
Will it have any impact or we are good with this?